### DISCUSSION TITLE
internal server error on mercadopago.com

### DISCUSSION THREAD

**Discussion by damianham** [mercadopago/sdk-js#54] [General]
I have been looking into integration of the SDK to create subscriptions.  While reading the documentation - the section relating to creating a token for a customer credit card referred to the web checkout integration page which contained examples of code to integrate a web form to capture customer card data to convert to a token.

After spending 1.5 days reading the documentation and integrating the form as per the documentation into a web page the image attached below was the result of the form submit.  This doesn't fill me with confidence to deploy this into production.  I think you have gone about this entirely the wrong way.  There is nothing wrong in having an SDK to make secure requests to your server APIs.  However a developer should be able to make individual requests at each stage of the process and have a much greater degree of control over the whole process.  

Rather than exposing a method to manipulate a web form e.g.
`
const mp = new MercadoPago('TEST-f0190789-1218-4810-9bb1-6d8280cd969a')
cardForm = mp.cardForm({
`
Why not expose the helper methods to convert the card number to the issuer, the valid document types, expiry date formatting etc.  Developers are entirely capable of building web forms that conform to their website style and using onChange handlers to handle data change events.  Developers are entirely capable of submitting requests to APIs and dealing with the results.  The SDK should expose methods to assist in doing so securely rather than be concerned with HTML form handling.

With epayco the process is
register a customer -> return customer ID
register a card for the customer -> returns a token
subscribe the customer to the service with the token -> returns a subscription ID
charge the subscription 

All POST requests with JSON encoded data, simple and reliable.  There are no issues with form styling or server errors, although epayco have other issues.

![Screenshot from 2022-02-22 16-05-48](https://user-images.githubusercontent.com/76936/155220704-126bd724-3a35-47b1-81a4-f2cbe69ffca3.png)

**Comment by lucmantovani** [comment]
Hello @damianham!
I am sorry you are facing problems while trying to integrate with our products.

We understand your point of providing the developers with more control of the whole process when desired.
This is why we do provide the core methods used internally to tokenize a credit card. You can learn more [here](https://www.mercadopago.com.co/developers/en/guides/online-payments/checkout-api/receiving-payment-by-card-core-methods).

Also, because the linked page is a guide to help to integrate as fast as possible, I recommend you to have a look at our technical documentation at this current repository, where we specify the methods with more detail.

The [API section](https://github.com/mercadopago/sdk-js#api) is where we describe the core methods that enable an integration in which you are able to control the whole process.
Since your issue happened specifically on trying to creating a credit card token, I believe the method you are most interested in is [createCardToken](https://github.com/mercadopago/sdk-js#mp-instancecreatecardtokencardtokenparams).

I hope it helps you!

Also, don't hesitate in contacting us again when needed.
